### PR TITLE
[FIX] hr_contract: fix tooltip warning

### PR DESCRIPTION
**Pull request by Bertrand2**
When the contract has a working schedule different from the employee's, a red triangle appears with a tooltip explaining the reason. That tooltip was implemented in the arch view using data-tooltip, which is not intended. This commit adds an owl widget replacing that issue.

In order to test that tooltip warning, it's only needed
to go on any running contract and change the resource_calendar
so it's different to the one on the employee's page.
Same process on the employee form.

### CODE REVIEW

File: addons/hr_contract/static/src/widgets/tooltip_warning_widget.xml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<templates id="template" xml:space="preserve">
+    <t t-name="hr_contract.ContractWarningTooltip" owl="1">
+        <span
+            class="fa fa-exclamation-triangle text-danger o_calendar_warning ms-3"
+            data-tooltip-template="hr_contract.CalendarMismatch"
+            data-tooltip-info='{
+                "text":"Calendar Mismatch : The employee&apos;s calendar does not match this contract&apos;s calendar. This could lead to unexpected behaviors."
```
Comment (kba-odoo) [inline addons/hr_contract/static/src/widgets/tooltip_warning_widget.xml:8]: I don't think the text will be translated this way.
Instead add a getter in the Component and return the Object with the text translated

File: addons/hr_contract/static/src/widgets/tooltip_warning_widget.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
+/** @odoo-module **/
+
+import { registry } from "@web/core/registry";
+
+const { Component } = owl;
+
+export class ContractWarningTooltip extends Component {}
+ContractWarningTooltip.template = "hr_contract.ContractWarningTooltip";
+
+registry.category("view_widgets").add("contract_warning_toolip", ContractWarningTooltip);
```
Comment (aab-odoo) [inline addons/hr_contract/static/src/widgets/tooltip_warning_widget.js:10]: typo toolip

File: addons/hr_contract/views/hr_contract_views.xml [hunk 3]
```diff
@@ -19,15 +19,15 @@
...
-                            </span>
+                        <div class="row">
+                            <div class="col-10">
+                                <field name="resource_calendar_id" required="1" nolabel="1"/>
+                            </div>
+                            <div class="col-2 pt-1">
+                                <widget
+                                    name="contract_warning_toolip"
+                                    attrs="{'invisible': [('calendar_mismatch', '=', False)]}"/>
+                            </div>
```
Comment (aab-odoo) [inline addons/hr_contract/views/hr_contract_views.xml:30]: why complexifying the arch? I'm not sure you're supposed to play with bootstrap cols
cc @adr-odoo

**Comment by adr-odoo** [comment]
Can I ask you to add the "steps to reproduce" in your commit message?

File: addons/hr_contract/views/hr_contract_views.xml [hunk 4]
```diff
@@ -188,16 +188,23 @@
...
+                                        <field name="resource_calendar_id"
+                                            groups="hr_contract.group_hr_contract_manager"
+                                            required="1"
+                                            nolabel="1"/>
+                                    </div>
+                                    <div class="col-2 pt-1">
+                                        <widget
+                                            name="contract_warning_toolip"
+                                            attrs="{'invisible': ['|', ('calendar_mismatch', '=', False), ('state', '!=', 'open')]}"/>
+                                    </div>
```
Comment (aab-odoo) [inline addons/hr_contract/views/hr_contract_views.xml:207]:
```xml
<div id="resource_calendar_warning" class="d-flex align-items-center">
    <field name="resource_calendar_id" groups="!hr_contract.group_hr_contract_manager" required="1" options="{'no_open': True, 'no_create': True}"/>
    <field name="resource_calendar_id" groups="hr_contract.group_hr_contract_manager" required="1"/>
    <widget name="contract_warning_toolip"/>
</div>
```

Comment (aab-odoo) [reply addons/hr_contract/views/hr_contract_views.xml:207]: Note that I removed the nolabels

**Review by aab-odoo — APPROVED** [review]

**Comment by kba-odoo** [comment]
robodoo r+

**Pull request closed without merging**